### PR TITLE
Utilize DF API in cols_to_array.

### LINKED ISSUE #5: cols_to_array does not support passing Columns
`cols_to_array` only accepts columns passed as a `str` and not as a `Column`. Use cases are when a user needs to pass a map with a specified key/value

```python
df = spark.createDataFrame(
    [
        (1, 2, {"a": 5}),
        (1, 2, {"a": 6}),
        (1, 2, {"b": 7})
    ],
    ["col1", "col2", "col3"]
)

df.withColumn("col4", cols_to_array("col1", F.col("col3")["a"]))
```

### PR DESCRIPTION
**Pull request by WesRoach**
Greatly simplified implementation.
Requires update to PySpark 3.1.0.

closes #5

### CODE REVIEW

**Comment by WesRoach** [comment]
@rileyschack Opening PR rather than commit to `main` as a quick heads-up that this requires updating to PySpark >=3.1.0.

File: pysoma/dataframe.py [hunk 1]
```diff
@@ -6,7 +6,9 @@
 from collections import namedtuple
 
 import pyspark.sql.functions as F
-from pyspark.sql import SparkSession, DataFrame, Column
+from pyspark.sql import SparkSession, DataFrame
+from pyspark.sql.column import Column
```
Comment (rileyschack) [inline pysoma/dataframe.py:10]: Any particular reason to move the `Column` import? No big deal either way

**Review by rileyschack — APPROVED** [review]
Just a general FYI, doesn't change the PR, but [PySpark 3.1.0 is not recommended for use](https://spark.apache.org/news/next-official-release-spark-3.1.1.html). Actually have to use PySpark 3.1.1 instead

File: pysoma/dataframe.py (hunk 1 above)
Comment (WesRoach) [reply pysoma/dataframe.py:10]: I had a brief implementation using PySpark `_func`s from the column module - I had lots of additional imports. Then I realized the DataFrame API now contains `filter` and none of that was required, removed them, just didn't move Column back - no hard opinion here either just YOLO'd it and left it on that line.

**Comment by WesRoach** [comment]
I hadn't realized that about `3.1.0`. Luckily conda-forge only has `3.1.1` which is what our internal env has been updated to. Good catch on that.

**Pull request merged**